### ISSUE TITLE
Map re-centres on browser resize

### ISSUE THREAD

**Issue by geoffbeaumont** [Joniff/Terratype#11]
When the browser window is resized, the Terratype map snaps back to the original centre - that's probably fine in the back office, but it's a problem in the front end if you have multiple markers, as the user may have already zoomed/panned to a new focus - for instance, I'm working on a map that has locations across England and Wales - the map starts out showing the whole area and centred on the midlands, and with several grouped markers. A user may well have drilled down into a group in their area (say Lancashire), so to throw them back to Birmingham is terrible usability...

I can see that if you have only one marker this behaviour is useful, but it could do with being switchable for front end rendering.

**Comment by geoffbeaumont** [comment]
We're using Google Maps provider, by the way - if that makes any difference!

**Comment by Joniff** [comment]
Agreed bug.

**Comment by mikaelkarlsson-se** [comment]
I have a similar problem where the map (Google Maps) most of the times changes position back to the one I've set to use on initiation, when zooming or clicking on one of the groups of markers.  
I guess it's the same bug, any idea when it might get solved @Joniff? This is a deal breaker in a customer project I'm working on, if we can use Terratype or not, so I really appreciate any feedback. Please tell if you need more information about the configuration used or any code examples.

**Comment by Joniff** [comment]
@mikaelkarlsson-se That does sound like a different bug, as the original bug report was when resizing the browser, which while I agree exists, I consider to be very low priority - because who spends all day resizing a window.

Your suggesting something different, which I have just quickly failed to replicate. I can click icons, pan around the map, click other icons and at no time does it reset to initial position, except when I resize the browser.

If you like to give me a better explanation of what is occurring for you, I might be able to further investigate.

**Comment by geoffbeaumont** [comment]
I can confirm that I'm *not* seeing the behaviour @mikaelkarlsson-se reported - however, I'm not passing in a position in the options, just setting a number of markers. Sounds like @mikaelkarlsson-se is passing in an explicit centre position?

**Comment by mikaelkarlsson-se** [comment]
Yes, I do set an explicit center position in the map options. The map works fine if I don't, and a few times it works also if I do.  
  
```
@Html.Terratype(new Options
{
    Provider = provider,
    MapSetId = 2,
    Height = 500,
    Zoom = 2,
    Position = new Terratype.CoordinateSystems.Wgs84("15,0")
})

foreach (var partner in mapModel.Partners)
    {
        if (partner.HasValue("location"))
        {
            var name = partner.GetVortoValue("partnerName");
            var url = partner.GetPropertyValue("websiteURL").ToString();
            var location = partner.GetPropertyValue<Terratype.Models.Model>("location");

            @Html.Terratype(
                            new Options
                            {
                                MapSetId = 2
                            }, location,
                @<text>
                    <h4>@name</h4>
                    @if (!string.IsNullOrEmpty(url))
                    {
                        <a href="@url" target="_blank">@url.Replace("http://", "").Replace("https://", "")</a>
                    }
                </text>
                        )
        }
    }
```  
  
I know that there are som JavaScript errors that might be of interest, but I can't get to the site at the moment. I will post them as well when I have the opportunity.

**Comment by mikaelkarlsson-se** [comment]
Sorry for the late reply @Joniff . I've checked for the JS errors I mentioned but it turned out that they where not related to this.

**Comment by Joniff** [comment]
New version 1.0.15 that tackles the issue of recentering when refreshing. There is now a new boolean field added to Options called AutoRecenterAfterRefresh, that controls whether you want a recenter or not. Default behaviour is off.

**Comment by mikaelkarlsson-se** [comment]
Hi @Joniff !
I've updated to the latest version and now Terratype ignores my custom position and instead always centers on the first marker in the map.  
My code is exactly the same as in my earlier comment. I've also experimented with the new options and they do not seem to affect/create my problem.  
  
Any ideas what could be wrong? :)

**Comment by Joniff** [comment]
Sorry about this @mikaelkarlsson-se, had a spot of regression. Thank you for reporting this. Have fixed it in the repository, but will have to wait until I do a release. Next week or so.

**Comment by mikaelkarlsson-se** [comment]
Sounds great @Joniff , thanks! :)

**Comment by mikaelkarlsson-se** [comment]
Hi @Joniff !
I've installed 1.0.16 but the problem still occurs, wasn't the fix included in that release?  
Also, I think I've read about it somewhere here, but the map takes quite some time to load since 1.0.15 (I think), how come and is it possible to make it work more "like before"? Sorry for going a bit OT. :)

**Comment by Joniff** [comment]
Hi @mikaelkarlsson-se 

Sorry, but this thread is getting too complicated to follow.

To recap

1. Someone else starts thread about an issue.
2. I agree its an issue
3. You post what looks like completely different bug
4. I make release, 1.0.15, that fixes original issue
4. I can't replicate your bug exactly, but do see something of tangent that might possible be your bug
5. I make release, 1.0.16, with fix for tangent bug
6. I'm not sure now, whether at anytime your bug has gone away. eg any of these releases has fixed your issue or if the bug has been present the whole time.
7. You ask me question about tweet you saw that Nik sent me. The next release, 1.0.17, will contain some code that will speedup the loading time of maps, I don't currently have a release date scheduled.

Sorry, but without further info, eg. A bug I can replicate, I'm unable to answer whether your issue has or hasn't been fixed now or in the past.

**Comment by mikaelkarlsson-se** [comment]
Sorry, I remembered wrong about this being the same bug or not as the reported.  
  
Hopefully the affected site will go live in a few days. When it has, I will create a new issue with more information. Thanks for your answers up until now and sorry for messing up this issue. :)

**Comment by mikaelkarlsson-se** [comment]
Sorry for spamming this even more @Joniff , just wanted to tell that I got the center issue solved after I found out that the Nuget update of the Google map provider had gone a bit wrong. Works now after a clean install. :)
